### PR TITLE
feat(FEC-11979): add setCurrentTime middleware

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

add setCurrentTime middleware
add support for middleware params chaining 

Solves FEC-11979

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [x] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [x] Docs have been updated

### CODE REVIEW

**Review by RoyBregman — APPROVED** [review]
LGTM, I suggest @semarche-kaltura will use this PR  locally and check if it works perfect for us before merging

File: src/middleware/middleware.js [hunk 1]
```diff
@@ -58,30 +58,36 @@ export default class Middleware {
...
    * @private
    * @returns {void}
    */
-  _executeMiddleware(middlewares: Array<Function>, callback: Function): void {
+  _executeMiddleware(middlewares: Array<Function>, callback: Function, params: Array<any>): void {
     const composition = middlewares.reduceRight(
       // eslint-disable-next-line no-unused-vars
       (next, fn) => v => {
-        fn(next);
+        params?.length > 0 ? fn(...params, next) : fn(next);
```
Comment (SivanA-Kaltura) [inline src/middleware/middleware.js:90]:
```suggestion
        params?.length ? fn(...params, next) : fn(next);
```

File: src/player.js [hunk 2]
```diff
@@ -2241,6 +2228,29 @@ export default class Player extends FakeEventTarget {
...
+  /**
+   * Set the current time in seconds.
+   * @param {Number} to - The number to set in seconds.
+   * @private
+   * @returns {void}
+   */
+  _setCurrentTime(to: number): void {
+    if (this._engine) {
+      if (Utils.Number.isNumber(to)) {
+        let boundedTo = to;
```
Comment (SivanA-Kaltura) [inline src/player.js:2240]:
```suggestion
        const boundedTo = to < 0 ? 0 : boundedTo;
```

File: src/player.js [hunk 3]
```diff
@@ -2241,6 +2228,29 @@ export default class Player extends FakeEventTarget {
...
+        if (to < 0) {
+          boundedTo = 0;
+        }
+        const safeDuration = this.isLive() ? this._engine.duration : this._engine.duration - DURATION_OFFSET;
+
+        if (boundedTo > safeDuration) {
+          boundedTo = safeDuration;
+        }
+        this._engine.currentTime = boundedTo;
```
Comment (SivanA-Kaltura) [inline src/player.js:2249]:
```suggestion
        this._engine.currentTime = Math.min(boundedTo, safeDuration);
```

File: src/player.js [hunk 4]
```diff
@@ -813,20 +813,7 @@ export default class Player extends FakeEventTarget {
...
-        }
-        const safeDuration = this.isLive() ? this._engine.duration : this._engine.duration - DURATION_OFFSET;
-
-        if (boundedTo > safeDuration) {
-          boundedTo = safeDuration;
-        }
-        this._engine.currentTime = boundedTo;
-      }
-    }
+    this._playbackMiddleware.setCurrentTime(to, this._setCurrentTime.bind(this, to));
```
Comment (SivanA-Kaltura) [inline src/player.js:816]: do you need to pass to to the bind as well, if it's already passed as a parameter to setCurrentTime ?

File: src/middleware/middleware.js (hunk 1 above)
Comment (SivanA-Kaltura) [reply src/middleware/middleware.js:90]: what is next in this case ? how does the middleware distinguish between next and params if it comes after params ?

File: src/middleware/middleware.js [hunk 5]
```diff
@@ -58,30 +58,36 @@ export default class Middleware {
...
    * @param {Function} callback - The callback function.
+   * @param {Array<any>} params - The action params.
    * @private
    * @returns {void}
    */
-  _executeMiddleware(middlewares: Array<Function>, callback: Function): void {
+  _executeMiddleware(middlewares: Array<Function>, callback: Function, params: Array<any>): void {
     const composition = middlewares.reduceRight(
       // eslint-disable-next-line no-unused-vars
       (next, fn) => v => {
```
Comment (SivanA-Kaltura) [inline src/middleware/middleware.js:89]:
```suggestion
      (next, fn) => () => {
```

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/middleware/middleware.js (hunk 1 above)
Comment (yairans) [reply src/middleware/middleware.js]: next have to be the last param

**Pull request merged**